### PR TITLE
Write up "What is Agile" and added Section Impact of AM on OS

### PR DESCRIPTION
**Pull request by smcdonou02**
- added new sources used
- wrote up section of "What is Agile" ==> need to add SCRUM and DSDM subsection 
- started write up of comparing management styles
- started the impact of Agile Methods on Open Source Development

### CODE REVIEW

File: main.typ [hunk 1]
```diff
@@ -38,31 +38,51 @@
 	instructor: "Harun Gultekin",
 )
 
+== Abstract 
+// do this last to summarize paper
 == Introduction
-
+// introduce the problem statement, our methods for finding papers, and what our paper will discuss
```
Comment (dyc3) [inline main.typ]: minor nit: so this is easily searchable later

```suggestion
// TODO: introduce the problem statement, our methods for finding papers, and what our paper will discuss
```

File: main.typ [hunk 2]
```diff
@@ -38,31 +38,51 @@
 	instructor: "Harun Gultekin",
 )
 
+== Abstract 
+// do this last to summarize paper
```
Comment (dyc3) [inline main.typ]: nit: same here

```suggestion
// TODO: do this last to summarize paper
```

File: main.typ [hunk 3]
```diff
@@ -38,31 +38,51 @@
...
 == Introduction
-
+// introduce the problem statement, our methods for finding papers, and what our paper will discuss
 == What is Agile?
+Agile Methodologies, while there are a multitude of strategies to follow, primarly have a shared goal of increasing the rate of development for development teams @Coram_2005. Software projects have many frequent changes throughout the life cycle. Developers needed a way to better handle these changes, which resulted in the creation of the Manifesto for Agile Software Development. The Manifesto emphasizes the value of the individual and interactions over tools and processes, and collaboration with the stakeholders over a negotiation. Functional software is also desired over solid documentation, as well as responsiveness and adaption to a situation rather than following a plan. @Eckstein_2004. To put it simply, the quality and efficiency of development is the goal of agile methods. Methods like Extreme Programming, SCRUM, Dynamic System Development Method (DSDM), and Test-Driven Development are popular in today's technology sector, all techniques designed for rapid development to deliver a product.
```
Comment (dyc3) [inline main.typ]: > Software projects have many frequent changes throughout the life cycle.

As a result of changing user requirements. 

The cause and effect relationship between waterfall development and agile is important to these guys. It's probably worth mentioning.

File: main.typ [hunk 4]
```diff
@@ -38,31 +38,51 @@
...
+
+=== Extreme Programming
+The objective of Extreme Programming (XP) is to get the project done in a straightforward manner. Compared to the highly decorative process of the waterfall model, XP relies on communication, feedback and simplicity @Coram_2005. XP works best for small, localized teams so the use of XP would not be benefical due to the nature of open source development.
+
+=== Test-Driven Development
+In Test-Driven Development, or TDD, requires tests to be written before the code itself is written. The design of the software changes through the tests as the rapid cycle of adding new tests, passing those tests and cleaning the code by refactoring is executed ("test-code-refactor") @Turnu_2006. It is imperative that the tests are automated to ensure that there is no technical debt from manually running tests. If they are automated, minimal effort is required to be run over and over again. The debugging process in TDD is simplified because bugs can be found earlier with the tests, making it easier and quicker to develop a full project @Turnu_2006.
```
Comment (dyc3) [inline main.typ]: nit: to be consistent with how abbreviations are presented I would prefer to start with:

> In Test-Driven Development (TDD), requires tests...

File: main.typ [hunk 5]
```diff
@@ -38,31 +38,51 @@
...
+Over the course of time, the genesis of open-source software has undergone notable evolution @Fitzgerald_2006. The initiation of an open-source project diverges markedly from that of a conventional software undertaking. In the latter, the project lifecycle encompasses four overarching phases: planning, analysis, design, and implementation. Conversely, certain open-source projects commence with a solitary developer or a small group who identify a personally compelling problem to address @Bonaccorsi_Rossi_2003. The planning phase is often condensed, and requirements analysis is frequently nominal or absent @Fitzgerald_2006. Notably, contemporary open-source projects have experienced increased financial backing, leading to a proliferation of startups embracing open-source development practices.
```
Comment (dyc3) [inline main.typ]: I just looked at the `cite` function docs again and I prefer this syntax. Didn't know typst let you cite things like this. But are these citations still broken for you? They work just fine for me and in CI.

Also, are you sure the citations are supposed to go in front of the sentence punctuation?

File: main.typ [hunk 6]
```diff
@@ -38,31 +38,51 @@
...
+Agile and Open Source management styles are quite similar in a few different aspects. Regarding the process of developing software, while agile methods have predefined processes and sequences, numerous open source projects also have structured processes @Eckstein_2004. Agile development requires close collaboration within the team (in methods of pair programming, review, etc), whereas open source teams are scattered in different parts of the world, working asyncronously. 
+// discuss customer/stakeholder interactions, early delievery and feedback, team structure
 
+== The Impact of Agile Methods on Open Source Development
+As Agile Methodologies integrate into mainstream development due to the positive effects on code quality and quick response time,aspects of agile are emerging into open source development. A trend in open source projects appear to be the adoption of Testing and Continuous Integration. The implementation of automated tests establishes trust among the developers involved with the open source project because passing tests demonstrates that their code works, and that the code does not have to be rewritten by other developers @Turnu_2006.
```
Comment (dyc3) [inline main.typ]: Honestly, yeah this is pretty good insight. Early open source projects didn't really do anything about CI and that's something I take for granted. The earliest commit from busybox I could find referencing testing was in 1999 https://git.busybox.net/busybox/commit/?id=a89910829c73c5ed4f30eecb181d7b3087ca4f74, which is significantly after most of the tools in busybox were introduced by bell labs.

**Review by dyc3 — CHANGES_REQUESTED** [review]
Looks good, mostly minor things I want clarification on.

File: main.typ (hunk 3 above)
Comment (smcdonou02) [reply main.typ]: facts

File: main.typ (hunk 4 above)
Comment (smcdonou02) [reply main.typ]: oo yes yes

File: main.typ (hunk 5 above)
Comment (smcdonou02) [reply main.typ]: I just looked at apa documentation it does go after the period, just feels so wrong.

Comment (smcdonou02) [reply main.typ]: also are you referring to @cite() or @name?

Comment (dyc3) [reply main.typ]: The `@name` syntax. (avoid accidentally @ ing people, it's considered bad manners). I'm fine either way, it's just syntax sugar, but if I was being pedantic, _technically_ its unrelated to the content changes in this PR. Ultimately that makes it harder to review. It's not something that matters to every team, but it's still something to consider.

**Comment by dyc3** [comment]
@smcdonou02 For my review comments, feel free to hit "resolve conversation" so that I know you've fixed it.

**Review by dyc3 — APPROVED** [review]

**Pull request merged**